src/main/java/com/brave/playlist/slidingpanel/SlidingPanelLayout.kt

```diff
@@ -0,0 +1,1000 @@
+        if (mPanelHeight == -1) {
+            mPanelHeight = (DEFAULT_PANEL_HEIGHT * density + 0.5f).toInt()
+        }
+        if (mShadowHeight == -1) {
+            mShadowHeight = (DEFAULT_SHADOW_HEIGHT * density + 0.5f).toInt()
+        }
+        if (mParallaxOffset == -1) {
+            mParallaxOffset = (DEFAULT_PARALLAX_OFFSET * density).toInt()
+        }
+        // If the shadow height is zero, don't show the shadow
+        mShadowDrawable = if (mShadowHeight > 0) {
+            ResourcesCompat.getDrawable(resources, if(mIsSlidingUp) R.drawable.above_shadow else R.drawable.below_shadow, null)
+        } else {
+            null
+        }
+        setWillNotDraw(false)
+        mDragHelper = ViewDragHelper.create(this, 0.5f, scrollerInterpolator, DragHelperCallback())
+        mDragHelper?.minVelocity = minFlingVelocity * density
+        mIsTouchEnabled = true
+    }
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+        if (mDragViewResId != -1) {
+            setDragView(findViewById(mDragViewResId))
+        }
+        if (mScrollableViewResId != -1) {
+            setScrollableView(findViewById(mScrollableViewResId))
+        }
+    }
+
+    fun setGravity(gravity: Int) {
+        require(!(gravity != Gravity.TOP && gravity != Gravity.BOTTOM)) { "gravity must be set to either top or bottom" }
+        mIsSlidingUp = gravity == Gravity.BOTTOM
+        if (!mFirstLayout) {
+            requestLayout()
+        }
+    }
+
+    var isTouchEnabled: Boolean
+        get() = mIsTouchEnabled && mSlideableView != null && mSlideState != PanelState.HIDDEN
+        set(enabled) {
+            mIsTouchEnabled = enabled
+        }
+
+    fun smoothToBottom() {
+        smoothSlideTo(0f, 0)
+    }
+
+    var panelHeight: Int
+        get() = mPanelHeight
+        set(value) {
+            if (panelHeight == value) {
+                return
+            }
+            mPanelHeight = value
+            if (!mFirstLayout) {
+                requestLayout()
+            }
+            if (panelState == PanelState.COLLAPSED) {
+                smoothToBottom()
+                invalidate()
+                return
+            }
+        }// Clamp slide offset at zero for parallax computation;
+
+    val currentParallaxOffset: Int
+        get() {
+            // Clamp slide offset at zero for parallax computation;
+            val offset = (mParallaxOffset * Math.max(mSlideOffset, 0f)).toInt()
+            return if (mIsSlidingUp) -offset else offset
+        }
+
+    fun addPanelSlideListener(listener: PanelSlideListener) {
+        synchronized(mPanelSlideListeners) { mPanelSlideListeners.add(listener) }
+    }
+
+    fun removePanelSlideListener(listener: PanelSlideListener) {
+        synchronized(mPanelSlideListeners) { mPanelSlideListeners.remove(listener) }
+    }
+
+    fun setDragView(dragView: View?) {
+        if (mDragView != null) {
+            mDragView!!.setOnClickListener(null)
```

Comment:
can we not just remove null check and use safe calls `?.` 
Something like this: `mDragView?.setOnClickListener(null)`